cmd/etcd.go

```diff
@@ -180,6 +185,20 @@ func RestoreEtcdSnapshot(
 	return APIURL, caCrt, clientCert, clientKey, certs, err
 }
 
+func validateETCDCerts(etcdHosts []*hosts.Host, state cluster.State) error {
+	if state.RancherKubernetesEngineConfig == nil {
+		// possibly already started a restore
+		return nil
+	}
+	for _, etcdHost := range etcdHosts {
+		certName := pki.GetCrtNameForHost(etcdHost, pki.EtcdCertName)
+		if cert := state.CertificatesBundle[certName].Certificate; cert == nil || cert.NotAfter.Before(time.Now().Add(time.Hour)) {
```

Comment:
should we check for presence of they key certName in the CertificatesBundle map?